File: projectePHP220917/Punt.php
<?php

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 * Description of Punt
 *
 * @author quim
 */
class Punt {

    //put your code here

    private $x;
    private $y;

//    function __construct() {
//        $punt = func_get_args();
//        if ($punt) {
//            $this->x = $punt['0'];
//
//            $this->y = $punt['1'];
//        } else {
//
//            $this->x = 0;
//
//            $this->y = 0;
//        }
//    }
//    
    function __construct($x, $y) {
        $this->x=$x;
        $this->y=$y;

    }
    
    public static function demanarPunt(){
        
        $x=0;
        $y=0;
        fscanf(STDIN, "%d\n", $x);
        fscanf(STDIN, "%d\n", $y);

        return new Punt($x,$y);
    }

    public function mostrar() {

        return "($this->x,$this->y)";
    }

    public function distanciaX($punt) {
        return ($punt->x) - ($this->x);
    }

    public function distanciaY($punt) {
        return ($punt->y) - ($this->y);
    }

    public function distanciaPunt($punt) {
        return Math . sqrt(Math . pow($this->distanciaX($punt), 2) + Math . pow($this->distanciaY($punt), 2));
    }

}

$obj = new Punt(1, 2);
$ob2 = new Punt(3, 4);

//echo $obj->mostrar();

//echo $obj->distanciaX($ob2);


$obj3 = new Punt(2,3);
$obj3::demanarPunt();
echo $obj3->mostrar();

File: perimetreRectangle/src/perimetrerectangle/Punt.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package perimetrerectangle;

import java.util.Scanner;

/**
 *
 * @author quim
 */
public class Punt {

    //les coordenades són reals per tant s'ha d'utilitzar float o double (double és mes efectiu).
    
    public double x;
    public double y;

    
    //per crear el Punt es necessita un constructor.
    
    public Punt() {
        
        //si no indiquem res es crea el Punt 0,0.
        this.x=0;
        this.y=0;
        
    }
    
    //es pot crear tots els constructors que vulguem.
    public Punt(double x, double y){
        this.x=x;
        this.y=y;
        
    }
    
    public double distanciaPunt(Punt p){
        return Math.sqrt(Math.pow(p.x-this.x,2)+Math.pow(p.y-this.y,2)); 
        
    }
    
    public double distanciaX(Punt p){
        return p.x - this.x; 
    }
    
    //Math.abs agafa el valor absolut sense el signe, si es negatiu retornara el valor sense el signe menys.
    
    public double distanciaY (Punt p){
        return p.y - this.y;
    }
    
    //
    public static Punt demanarPunt (){
        Scanner entrada = new Scanner(System.in);
        System.out.println("Digues un punt (x,y):");
        double x=entrada.nextDouble();
        double y=entrada.nextDouble();
        return new Punt(x,y);
    }
    
    //aquest parametre retorna el que nosaltres volem que retorni.
    
    public String toString(){
        
        return "(" + this.x + "," + this.y+ ")";
        
    }
    
    
    
    
     public static void main(String[] args) {
         
         Punt p1 = Punt.demanarPunt();
         Punt p2= Punt.demanarPunt();
         
         System.out.println(p1.distanciaX(p2));
         System.out.println(p1.distanciaY(p2));
         
         
         
         
         
         
         
         
         
         
         
         
         /*
         Punt p = new Punt();
         
         System.out.println(p);
         
         Punt p1 = new Punt(4,4);
         
         System.out.println(p.distanciaPunt(p1));
         
         
         System.out.println(p.distanciaX(p1));
         
         
         System.out.println(p.distanciaY(p1));
         */
     }
    
}
